### ISSUE TITLE
Products not visible when user is in an owner Role in a Security Group

### ISSUE THREAD

**Issue by ggraynoth** [salesagility/SuiteCRM#10026]
#### Issue
This issue appears to relate specifically to the Product module only. 
We have set up a number of Security Groups - ****Procurement and for each there are two roles set up. One for the Managers, where the Role is set to Group visibility for all modules and a Support role where the role is set to Owner for all modules.

#### Expected Behavior
So, when a Support user creates a product (which they can) they should be able to see them but not any other data.
Their Manager should be able to see their data and also that of any other Support user that is in the Group.

#### Actual Behavior
This works fine for all module EXCEPT Products. 
The User can create a product and it is saved OK. 
However, when they go into List view, the list is blank.

Their Manager can see the products created and use them as required and can also select them and do a Mass Update to assign them to the Support user , who can then see them.
 
#### Possible Fix
We applied a suggested fix to SecurityGroup.PHP but this has not resolved the problem.

It appears that upon creation the Assigned user_id field in the database is not being set, as the value is NULL for records created by the Support user.

#### Steps to Reproduce
Create a Security group
Create Roles as described.
Add users into each Role
Create a product as the Owner type user and try to view Products - it will be blank. then login as the manager  type user in the same group and you will see the products created.

We have checked the database and after creation, the assigned_user_id of the products created is NULL. After they have been assigned by the manager using a mass update, they are set to the owner users assigned id. So, the field is not being set correctly on new Product creation.

#### Context
We have had to allow the support users to move to the Group Role and use filters to see only their product categories. the Owner role would be perfect if it worked.

PRIORITY Moderate as there is a work around of sorts.

#### Your Environment

* SuiteCRM Version used: 4.2.3
* Browser name and version  : Chrome - Version 112.0.5615.137 (Official Build) (64-bit)
* Environment name and version : MySQL v8.0.2, PHP 7.4.33, Apache 2.4.54
* Operating System and version : Windows

**Comment by ggraynoth** [comment]
Priority on this is moderate as there is a workaround of sorts.

**Comment by ggraynoth** [comment]
On further testing - checking the assigned_user _id after changing various security group and owner role settings, the assigned_user_id is always set to NULL and only changed to the Owner user when a manager does a mass update to assign them. 
Also, if you set the products module setting in the Role to group for list an view, the user sees all of the products in the group. However, using product filters in the list view, if you filter just on 'Created By' and select the Owner user who created them, it only shows those records that have been assigned via the mass update., even through all of the records created contain the Owner user's correct if in the Created By fieeld in the database. this brings us back to there being a check against the assigned_user_id somewhere  which is suppressing these from being displayed.

This looks like it has to be in the security module but struggling to see how this can be fixed.

**Comment by ggraynoth** [comment]
Problem solved. 

Feedback in the SuiteCRM forum pointed us in the right direction and there is a very simple, non-coding solution.
In Studio, we added the 'Assigned to' field onto the Product Edit and Detail view and saved and deployed them. We added it at the bottom of the panel, alongside the product image field, which we don't use anyway.

This now comes up on the screen, auto-filled with the creators user ID and on save, the assigned_user_id in the dB is written with the correct value. The creating user can then see these in list view and everything works as expected.

We were looking far too deeply, into code, for the solution, which in reality, as is often the case, was a simple configuration change to fix.

Possibly something that can be configured as standard or included in the documentation, to prevent anyone doing what we are doing spending a lot of unnecessary time finding this out. It is also documented now in the SuiteCRM Forum.
